ts/components/messages/MessageCTABar.tsx

```diff
@@ -141,68 +141,101 @@ class MessageCTABar extends React.PureComponent<Props, State> {
     return this.props.payment !== undefined;
   }
 
+  private searchEventInCalendar = async (
+    endDate: Date,
+    title: string
+  ): Promise<string | undefined> => {
+    const startDate = new Date(endDate.getTime());
+    const isEventInCalendar = RNCalendarEvents.fetchAllEvents(
+      formatDateAsReminder(new Date(startDate.setDate(endDate.getDate() - 1))),
+      formatDateAsReminder(endDate)
+    )
+      .then(
+        events => {
+          if (events) {
+            const event = events.find(e => {
+              return (
+                e.title === title &&
+                new Date(e.endDate).getDay() === endDate.getDay()
+              );
+            });
+            if (event) {
+              return event.id;
+            }
+          }
+          return undefined;
+        },
+        // handle promise rejection
+        () => {
+          return undefined;
+        }
+      )
+      .catch();
+
+    return await isEventInCalendar;
+  };
+
   /**
    * A function to check if the eventId of the CalendarEvent stored in redux
    * is really/still in the device calendar.
    * It is important to make this check because the event can be removed outside
    * the App.
    */
-  private checkIfEventInCalendar = (calendarEvent: CalendarEvent) => {
-    checkAndRequestPermission()
-      .then(
-        hasPermission => {
-          if (hasPermission) {
-            RNCalendarEvents.findEventById(calendarEvent.eventId)
-              .then(
-                event => {
-                  if (event) {
-                    // The event is in the store and also in the device calendar
-                    // Update the state to display and handle the reminder button correctly
+  private checkIfEventInCalendar = (
+    calendarEvent: CalendarEvent | undefined
+  ) => {
+    if (calendarEvent) {
```

Comment:
```suggestion
    if (calendarEvent === undefined) {
      this.setState({
        isEventInDeviceCalendar: false
      });
      return;
    }
```
you could return immediately if calendar is undefined